### ISSUE TITLE
Ad Hoc polling for background task status broken since 2.0.0.0

### ISSUE THREAD

**Issue by geerlingguy** [ansible/ansible#61015]
##### SUMMARY

In Ansible for DevOps, I wrote a section on async and background tasks, and in it, I mentioned the option of setting a `-P` 'POLL_INTERVAL' to allow Ansible to print the status of the backgrounded job (when used with `-B`) every X seconds (15 seconds by default).

In issue https://github.com/geerlingguy/ansible-for-devops/issues/91 people were reporting issues with this for quite some time, since early 2018. It looks like one of the primary issues—not printing the background task job ID—has been fixed in https://github.com/ansible/ansible/commit/df6b8d2a4a91e59d40bfb7c2b8b80bad48cd4585

However, I was puzzled as to why polling seems to do nothing. I decided to basically keep installing older versions of Ansible until it _did_ work, and I installed every point release in the 2.x series, and none of them work (instead of printing polling information, it just sits with no output on the command line). In 1.9.x, though, I get polling output every `-P` seconds (15 by default).

The documentation for [Time Limited Background Operations](https://docs.ansible.com/ansible/latest/user_guide/intro_adhoc.html#time-limited-background-operations) has not changed significantly since 1.x, nor have the descriptions of the CLI options. Therefore I presume this is a bug, as I never remember seeing any changelog release notes about live polling status updates being removed from the CLI when running ad-hoc commands.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
ad-hoc command background operation polling

##### ANSIBLE VERSION

```paste below
ansible 2.9.0.dev0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/Users/jgeerling/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/jgeerling/Downloads/ansible/lib/ansible
  executable location = /Users/jgeerling/Downloads/ansible/bin/ansible
  python version = 2.7.16 (default, Apr 12 2019, 15:32:40) [GCC 4.2.1 Compatible Apple LLVM 10.0.1 (clang-1001.0.46.3)]
```

##### CONFIGURATION

```paste below
ANSIBLE_NOCOWS(/etc/ansible/ansible.cfg) = True
ANSIBLE_PIPELINING(/etc/ansible/ansible.cfg) = True
ANSIBLE_SSH_CONTROL_PATH(/etc/ansible/ansible.cfg) = /tmp/ansible-ssh-%%h-%%p-%%r
DEFAULT_FORKS(/etc/ansible/ansible.cfg) = 20
DEFAULT_HOST_LIST(/etc/ansible/ansible.cfg) = [u'/etc/ansible/hosts']
DEFAULT_ROLES_PATH(/etc/ansible/ansible.cfg) = [u'/Users/jgeerling/Dropbox/VMs/roles']
DEFAULT_STDOUT_CALLBACK(/etc/ansible/ansible.cfg) = yaml
RETRY_FILES_ENABLED(/etc/ansible/ansible.cfg) = False
```

##### OS / ENVIRONMENT
macOS 10.14, target OS either Debian 9/10 or Ubuntu 16.04

##### STEPS TO REPRODUCE

    ansible all -b -P 10 -B 3600 -m shell -a "while true; do echo test; sleep 5; done"

##### EXPECTED RESULTS

Every 10 seconds, Ansible should output status information, for example, this is what is printed every 10 seconds in Ansible 1.9.1 running the above command:

```
192.168.60.5 | success >> {
    "ansible_job_id": "363390903939.6230", 
    "changed": false, 
    "finished": 0, 
    "results_file": "/root/.ansible_async/363390903939.6230", 
    "started": 1
}

<job 363390903939.6230> polling on 192.168.60.5, 3590s remaining
```

##### ACTUAL RESULTS

No output was given whatsoever. After running the command, it just sits indefinitely until the background timeout (3600 seconds) expires, or the command completes (in this case, since it's an infinite `while` loop it never will). At that point, it prints the job and status information, but only once, for everything.

Related issues:

  - https://github.com/geerlingguy/ansible-for-devops/issues/91

**Comment by ansibot** [comment]
Files identified in the description:
None

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by sivel** [comment]
Duplicate of https://github.com/ansible/ansible/issues/14681
